### PR TITLE
[TIR] Enhance RemoveNoOp pass to remove negative loop

### PR DESCRIPTION
**Pull request by yincs-intellif**
When after partition cascade ops, we sometimes meet some unused loops because their extents are negative. 
This PR enhance RemoveNoOp pass to remove these unused loop.

cc @Hzfengsy

### CODE REVIEW

File: src/tir/transforms/remove_no_op.cc [hunk 1]
```diff
@@ -162,6 +168,9 @@ class NoOpRemover : public StmtMutator {
     }
     return stmt.defined() ? stmt : Evaluate(0);
   }
+
+  std::unordered_map<const VarNode*, arith::IntSet> var_range_map;
+  arith::Analyzer analyzer;
```
Comment (Hzfengsy) [inline src/tir/transforms/remove_no_op.cc:173]:
```suggestion
  std::unordered_map<const VarNode*, arith::IntSet> var_range_map_;
  arith::Analyzer analyzer_;
```
Also, please add the document

**Review by Hzfengsy — APPROVED** [review]
Overall LGTM

File: src/tir/transforms/remove_no_op.cc [hunk 2]
```diff
@@ -87,9 +87,15 @@ class NoOpRemover : public StmtMutator {
     }
   }
   Stmt VisitStmt_(const ForNode* op) final {
+    var_range_map_[op->loop_var.get()] = arith::IntSet::Interval(op->min, op->min + op->extent - 1);
```
Comment (wrongtest-intellif) [inline src/tir/transforms/remove_no_op.cc:90]: recommend arith::IntSet::FromMinExtent() :)

**Review by wrongtest-intellif — CHANGES_REQUESTED** [review]

File: src/tir/transforms/remove_no_op.cc (hunk 2 above)
Comment (yincs-intellif) [reply src/tir/transforms/remove_no_op.cc]: OK

File: src/tir/transforms/remove_no_op.cc (hunk 1 above)
Comment (yincs-intellif) [reply src/tir/transforms/remove_no_op.cc]: OK

**Review by wrongtest-intellif — APPROVED** [review]

**Pull request merged**